spec/controllers/application_controller_spec.rb

```diff
@@ -532,8 +537,9 @@ def index
       expect(mixpanel_spy).to have_received(:run).with(
         distinct_id: "123",
         event_name: "beep",
-        data: expected_mixpanel_data
+        data: anything
```

Comment:
What's the deal with this change? Not a blocker. (I know there's the extra `expect` on line 542, but what's the reason for it?)

Reply:
The `data:` changed subtly (device_brand changed from `nil` to `"Apple"`) but the change was impossible to identify in the output of `expect(..).to have_received(...).with` because the output is on a single line and truncated.

If we do an independent expectation just on the `data` param (which AFAIK RSpec makes inconvenient to do, you have to do something like the arg capture strategy I'm using above) RSpec makes a sensible diff.

really I've never liked `expect ... have_received ... with` and wish RSpec could be configured to capture the call args on spies and let you do something like `expect(spy).calls_to(:foo).first[:data].to eq(expected_data)`

but this is possibly a personal issue